### PR TITLE
consul with docker access

### PR DESCRIPTION
**Pull request by awangc**
- consul container's entry point downgrades the running container's user to `consul:consul` (https://github.com/hashicorp/docker-consul/blob/master/0.X/docker-entrypoint.sh#L85). This can lead to a mismatch of the GID between the user `consul` and `/var/run/docker.sock`, resulting in permission denied if the container mounts `/var/run/docker.sock` from the host and invokes `docker`. We modify the original entry point function to make sure user consul belongs to the GID of `/var/run/docker.sock` 

- upgrades versions for consul and consul-esm

### CODE REVIEW

File: consul/consul-entrypoint-docker.sh [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+fi
+
+# If the user is trying to run Consul directly with some arguments, then
+# pass them to Consul.
+if [ "${1:0:1}" = '-' ]; then
+    set -- consul "$@"
+fi
+
+# Look for Consul subcommands.
+if [ "$1" = 'agent' ]; then
```
Comment (zeeshanlakhani) [inline consul/consul-entrypoint-docker.sh]: minor: maybe make 'agent', 'version' strings vars up top or before these calls?

File: consul/Dockerfile [hunk 2]
```diff
@@ -1,7 +1,11 @@
-FROM consul:1.6.2
+FROM consul:1.6.3
 
 LABEL maintainer="williamofockham <occam_engineering@comcast.com>"
 
 RUN apk update \
-  && apk add docker \
+  && apk add docker shadow \
```
Comment (zeeshanlakhani) [inline consul/Dockerfile]: shadow?

Comment (awangc) [reply consul/Dockerfile]: This package provides `groupmod` and `usermod` in alpine

File: consul/consul-entrypoint-docker.sh (hunk 1 above)
Comment (awangc) [reply consul/consul-entrypoint-docker.sh]: well, right now these are the only places where we have check against `'agent'` and `'version'`. i think if later we reference them at multiple locations then we can make them into vars. Plus this script is based on the original from `docker-consul`. Maybe upstream there if we feel it should be changed?

**Comment by zeeshanlakhani** [comment]
🌮

File: docker.mk [hunk 3]
```diff
@@ -45,3 +45,4 @@ push-all:
 	@make -f tcpreplay/docker.mk push
 	@make -f dpdk/docker.mk push
 	@make -f sandbox/docker.mk push
+
```
Comment (zeeshanlakhani) [inline docker.mk]: new line needed?

Comment (awangc) [reply docker.mk]: no, I'll remove it.

**Pull request merged**